packages/pyolite-kernel/py/pyolite/pyolite/kernel.py

```diff
@@ -1,14 +1,17 @@
 # This is our ipykernel mock
 from ipykernel import CommManager
 
-from .display import display_publisher
+from .custom_display import XDisplayHook, XDisplayPublisher
 from .interpreter import Interpreter
-from .custom_display import XDisplayHook
+
 
 class Pyolite:
     def __init__(self):
-        self.interpreter = Interpreter(kernel=self, displayhook_class=XDisplayHook)
```

Comment:
You need to pass the `display_pub_class` to the Shell constructor

Reply:
Ah. Commenting old code, sorry.

Reply:
Yes I did, this is an outdated file